### ISSUE TITLE
Clarification Needed for Oxen Token Claim Process

### ISSUE THREAD

**Issue by venezuela01** [loki-project/loki-improvement-protocols#64]
**Clarification Needed for Oxen Token Claim Process**

> Oxen Token Claims
> Holders may use their Oxen to claim new Session Tokens. The amount of Session Tokens claimed is proportional to the total Oxen circulating supply at the time of the claim. For example, if a holder burnt 1% of the circulating supply of Oxen, they would receive 1% of the Session Tokens allocated to Oxen Token Claims (300,000).

The statement above requires clarification due to potential ambiguities or issues.

Firstly, 'circulating supply' needs a precise definition. If 1 million Oxen tokens are burned and converted to a Session Token, then should the Oxen's current circulating supply be reduced by 1 million? Are we in agreement on this?

Secondly, the phrase "receiving 1% of the Session Tokens allocated to Oxen Token Claims (300,000)" needs more detailed explanation.

Consider this scenario:

The initial supply of Oxen is 66 million. The total number of Session Tokens in the claim pool is 30 million. Suppose two hypothetical 'whales' each hold 33 million Oxen. If the first whale burns 33 million Oxen and claims 15 million Session Tokens, and the second whale then burns their 33 million Oxen (now representing 100% of the remaining supply), should the second one be entitled to 100% of the 30 million Session Tokens, which is 30 million Session Tokens, or just 50% of 30 million Session Tokens, which is 15 million Session Tokens?

Clearly, the former is not what the team implies. This statement needs careful wording because the Oxen's circulating supply is variable and the remaining Session claiming pool is also variable. Without an agreement on a timestamp to snapshot the supply, there is no fixed amount of supply.

A more precise statement might be:

**The amount of Session Tokens claimed is proportional to the total Oxen circulating supply at the time of the claim, and also proportional to the size of the remaining unclaimed Session Tokens in the pool.**

```
# Burning Percentage of Circulation Supply
burning_percentage_of_circulation_supply = (burning_oxen_token / circulation_supply_before_burning) * 100

# Expected Swapped Session Token
expected_swapped_session_token = (burning_oxen_token / circulation_supply_before_burning) * remaining_session_token_pool_before_burning

# Swap Ratio
swap_ratio = expected_swapped_session_token / burning_oxen_token
```

For example:

| Timestamp | Circulation Supply Before Burning | Burning Oxen Token | Accumulated Burnt Oxen | Burning Percentage of Circulation Supply | Remaining Session Token Pool Size Before Burning | Expected Swapped Session Token | Dynamic Swap Ratio |
|:---------:|----------------------------------:|-------------------:|-----------------------:|-----------------------------------------:|-------------------------------------------------:|------------------------------:|-------------------:|
| T1        |                          66,000,000 |          6,000,000 |              6,000,000 |                                  9.09091 |                                      30,000,000 |                2,727,270 |           0.454545 |
| T2        |                          60,000,000 |         10,000,000 |             16,000,000 |                                 16.6667  |                                      27,272,700 |                4,545,450 |           0.454545 |
| T3        |                          50,000,000 |          8,000,000 |             24,000,000 |                                       16 |                                      22,727,300 |                3,636,360 |           0.454545 |
| T4        |                          42,000,000 |          6,000,000 |             30,000,000 |                                 14.2857  |                                      19,090,900 |                2,727,270 |           0.454545 |
| T5        |                          36,000,000 |          4,000,000 |             34,000,000 |                                 11.1111  |                                      16,363,600 |                1,818,180 |           0.454545 |
| T6        |                          32,000,000 |          2,000,000 |             36,000,000 |                                      6.25 |                                      14,545,500 |                  909,091 |           0.454545 |

It's important to document this process clearly so we won't make stupid mistakes in the smart contract. If the specification is unclear, you can't expect the smart contract to be safe.

**Comment by venezuela01** [comment]
@KeeJef

**Comment by KeeJef** [comment]
Yes, ill see if I can get the team to update this, in the blog. 

> The initial supply of Oxen is 66 million. The total number of Session Tokens in the claim pool is 30 million. Suppose two hypothetical 'whales' each hold 33 million Oxen. If the first whale burns 33 million Oxen and claims 15 million Session Tokens, and the second whale then burns their 33 million Oxen (now representing 100% of the remaining supply), should the second one be entitled to 100% of the 30 million Session Tokens, which is 30 million Session Tokens, or just 50% of 30 million Session Tokens, which is 15 million Session Tokens?

So in this scenario both swappers would receive 15 million Session tokens, that's what were shooting for here. Since both burned 50% of the total Oxen supply they would split the swap pool 50/50. However i see the confusion here when we consider that this burning happens over time, with the pool reducing in size instead of having a discrete event.

Ignoring the swap multiplier, the calculation to determine how many Session tokens a user would get, if they purely paricipated in the Oxen swap, would be 

(Coins burned by user / Snapshotted Oxen supply at Session token TGE ) * 30,000,000

> Clearly, the former is not what the team implies. This statement needs careful wording because the Oxen's circulating supply is variable and the remaining Session claiming pool is also variable. Without an agreement on a timestamp to snapshot the supply, there is no fixed amount of supply.

On this I expect we will snapshot the Oxen supply on the Session token TGE date
